src/sinks/kafka/config.rs

```diff
@@ -159,79 +157,73 @@ impl KafkaSinkConfig {
 
         self.auth.apply(&mut client_config)?;
 
-        match kafka_role {
```

Comment:
Note that I only removed the `KafkaRole::Consumer` branch in this diff. The body of the `KafkaRole::Producer` branch remains unchanged.